core/trino-main/src/main/java/io/trino/operator/scalar/ArrayHistogram.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar;
+
+import io.trino.operator.aggregation.histogram.SingleTypedHistogram;
+import io.trino.operator.aggregation.histogram.TypedHistogram;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.MapBlock;
+import io.trino.spi.function.Convention;
+import io.trino.spi.function.Description;
+import io.trino.spi.function.OperatorDependency;
+import io.trino.spi.function.OperatorType;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlType;
+import io.trino.spi.function.TypeParameter;
+import io.trino.spi.type.MapType;
+import io.trino.spi.type.Type;
+import io.trino.type.BlockTypeOperators;
+
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.BLOCK_POSITION;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.FAIL_ON_NULL;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.NULLABLE_RETURN;
+
+@Description("Return a map containing the counts of the elements in the array")
+@ScalarFunction(value = "array_histogram")
+public final class ArrayHistogram
+{
+    private ArrayHistogram() {}
+
+    @TypeParameter("T")
+    @SqlType("map(T, BIGINT)")
+    public static Block arrayHistogram(
+            @TypeParameter("T") Type elementType,
+            @OperatorDependency(
+                    operator = OperatorType.EQUAL,
+                    argumentTypes = {"T", "T"},
+                    convention = @Convention(arguments = {BLOCK_POSITION, BLOCK_POSITION}, result = NULLABLE_RETURN))
+            BlockTypeOperators.BlockPositionEqual equalOperator,
+            @OperatorDependency(
+                    operator = OperatorType.HASH_CODE,
+                    argumentTypes = "T",
+                    convention = @Convention(arguments = BLOCK_POSITION, result = FAIL_ON_NULL))
+            BlockTypeOperators.BlockPositionHashCode hashCodeOperator,
+            @TypeParameter("map(T, BIGINT)") MapType mapType,
+            @SqlType("array(T)") Block arrayBlock)
+    {
+        TypedHistogram histogram = new SingleTypedHistogram(elementType, equalOperator, hashCodeOperator, 10);
+        int positionCount = arrayBlock.getPositionCount();
+        for (int position = 0; position < positionCount; position++) {
+            if (!arrayBlock.isNull(position)) {
```

Comment:
Do you agree with my choice to ignore nulls? The alternatives would be to either include nulls as a key in the output map with their respective count (assuming null can actually be a key in a map) or to return null as the result of the function if any element in the array is null.

Reply:
I'd say this is a pretty fair assumptions given that's how most of the [other aggregate operators work](https://trino.io/docs/current/functions/aggregate.html).

> Except for [count()](https://trino.io/docs/current/functions/aggregate.html#count), [count_if()](https://trino.io/docs/current/functions/aggregate.html#count_if), [max_by()](https://trino.io/docs/current/functions/aggregate.html#max_by), [min_by()](https://trino.io/docs/current/functions/aggregate.html#min_by) and [approx_distinct()](https://trino.io/docs/current/functions/aggregate.html#approx_distinct), all of these aggregate functions ignore null values and return null for no input rows or when all values are null. For example, [sum()](https://trino.io/docs/current/functions/aggregate.html#sum) returns null rather than zero and [avg()](https://trino.io/docs/current/functions/aggregate.html#avg) does not include null values in the count. The coalesce function can be used to convert null into zero.

Reply:
Just found the code which does not allow nulls as a key in maps in Trino: https://github.com/trinodb/trino/blob/19f61fc56984bd98c617d8e72214a61237f8b256/core/trino-main/src/main/java/io/trino/operator/scalar/MapConstructor.java#L135